phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -4588,4 +4424,195 @@ private TableName getParentPhysicalTableName(PTable table) {
+                                cells = m.getFamilyCellMap().get(familyName);
+                            }
+
+                            for (Cell cell : cells) {
+                                if (Bytes.compareTo(CellUtil.cloneQualifier(cell), PhoenixDatabaseMetaData.DATA_TYPE_BYTES) == 0) {
+                                    dataTypeCell = cell;
+                                } else if (Bytes.compareTo(CellUtil.cloneQualifier(cell), PhoenixDatabaseMetaData.COLUMN_SIZE_BYTES) == 0) {
+                                    columnSizeCell = cell;
+                                } else if (Bytes.compareTo(CellUtil.cloneQualifier(cell), PhoenixDatabaseMetaData.DECIMAL_DIGITS_BYTES) == 0) {
+                                    decimalDigitCell = cell;
+                                }
+                            }
+                        }
+
+                        // After PHOENIX-3534, we don't store parent table column metadata along with the child metadata,
+                        // so we don't need to propagate changes to the child views.
+
+                        if (!table.getIndexes().isEmpty()) {
+                            PhoenixConnection connection = null;
+                            try {
+                                connection = QueryUtil.getConnectionOnServer(env.getConfiguration()).unwrap(PhoenixConnection.class);
+                            } catch (ClassNotFoundException e) {
+                            }
+
+                            for (PTable index : table.getIndexes()) {
+                                byte[] tenantIdBytes = index.getTenantId() == null ?
+                                                ByteUtil.EMPTY_BYTE_ARRAY :
+                                                index.getTenantId().getBytes();
+                                byte[] schemaNameBytes = index.getSchemaName().getBytes();
+                                byte[] indexName = index.getTableName().getBytes();
+                                byte[] indexKey = SchemaUtil.getTableKey(tenantIdBytes, schemaNameBytes, indexName);
+
+                                IndexMaintainer indexMaintainer = index.getIndexMaintainer(table, connection);
+                                boolean isColumnIndexed = indexMaintainer.getIndexedColumnInfo().contains(
+                                                new Pair<>(column.getFamilyName().getString(), column.getName().getString()));
+                                ColumnReference coveredColumn = indexMaintainer.getCoveredColumnsOfIndexTable(
+                                                new ColumnReference(column.getFamilyName().getBytes(), column.getColumnQualifierBytes()));
+
+                                // Since the columns of fixed length which in present in primary key of index table will be converted
+                                // to variable length when index tables are created, So we will not process indexed columns.
+                                if (isColumnIndexed) {
```

Comment:
remove this boolean as its not used, just keep the comment about why we don't need to do anything for indexed columns.

Reply:
Ok